qfieldsync/core/offline_converter.py

```diff
@@ -155,6 +155,12 @@ def convert(self):
                             'Both "Area of Interest" and "only selected features" options were enabled, tha latter takes precedence.'),
                             'QFieldSync')
                     self.__offline_layers.append(layer)
+
+                    # Store the primary key field name(s) as comma separated custom property
+                    if layer.type() == QgsMapLayer.VectorLayer:
+                        key_fields = ','.join([layer.fields()[x].name() for x in layer.primaryKeyAttributes()])
```

Comment:
There is this very ugly situation when the column name could store the list separator. In this case a column with `"crazy,comma"` would fail. I think it would be better to directly give me the single column I would better use, instead of the full list. QField ignores the rest of the columns anyway:
```suggestion
                        key_field = [*layer.primaryKeyAttributes(), 'fid'][0]
                        layer.setCustomProperty('QFieldSync/cloudPrimaryKey', key_field)
```

Reply:
But what happens in the case of a pk on more than one field? It's not better to check the result on the QField side for multiple values and show an error like "I cannot work with multi pks

And isn't more clear to not set the value at all (or empty string) when no pk is defined, instead of a harcoded 'fid' value, that can be wrong?

Reply:
But what happens in the case of a pk on more than one field? It's not better to check the result on the QField side for multiple values and show an error like "I cannot work with multi pks"?

And isn't more clear to not set the value at all (or empty string) when no pk is defined, instead of a harcoded 'fid' value, that can be wrong?

Reply:
Multiple primary keys are a problem, true that. Currently I use the first column anyway, but I can also complain, not a big difference to me.

Assuming an attribute callled "fid" is not the best solution, but it works on memory and geojson layers, otherwise I cannot write tests.

I am ok to keep your original implementation, just add an assert that the column names do not contain the delimiter (`,` in this case).

Reply:
I can also use a different delimiter than `,`...

Reply:
Is there something that cannot be in the field name?

Reply:
AFAIK in postgres there is no limitation what symbols can be used as a column name